pkg/scan/apk.go

```diff
@@ -141,15 +135,18 @@ func scan(s *sbomSyft.SBOM, localDBFilePath string) (*Result, error) {
 
 	matches := matchesCollection.Sorted()
 
-	var findings []*Finding
+	var findings []Finding
 	for i := range matches {
 		m := matches[i]
 
 		finding, err := mapMatchToFinding(m, datastore)
 		if err != nil {
 			return nil, fmt.Errorf("failed to map match to finding: %w", err)
 		}
-		findings = append(findings, finding)
+		if finding == nil {
```

Comment:
Not today, but wanted to be safe here to avoid a panic in the future from changes to the mapping logic, since we're dereferencing the pointer immediately after here